src/Azure/Orleans.Streaming.EventHubs/Providers/Streams/EventHub/EventHubStreamBuilder.cs

```diff
@@ -1,14 +1,57 @@
+using System;
+using Microsoft.Extensions.Options;
 using Microsoft.Extensions.DependencyInjection;
 using Orleans.Configuration;
 using Orleans.ServiceBus.Providers;
-using System;
-using Microsoft.Extensions.Options;
 using Orleans.Providers.Streams.Common;
 using Orleans.ApplicationParts;
 
 namespace Orleans.Streams
 {
-    public class SiloEventHubStreamConfigurator : SiloRecoverableStreamConfigurator
+    public interface ISiloEventHubStreamConfigurator : ISiloRecoverableStreamConfigurator
+    {
+    }
+
+    public static class SiloEventHubStreamConfiguratorExtensions
+    {
+        public static TConfigurator ConfigureCheckpointer<TConfigurator,TOptions>(this TConfigurator configurator, Func<IServiceProvider, string, IStreamQueueCheckpointerFactory> checkpointerFactoryBuilder, Action<OptionsBuilder<TOptions>> configureOptions)
+            where TConfigurator : ISiloEventHubStreamConfigurator
+            where TOptions : class, new()
+        {
+            configurator.ConfigureComponent<TOptions, IStreamQueueCheckpointerFactory>(checkpointerFactoryBuilder, configureOptions);
+            return configurator;
+        }
```

Comment:
I take it users aren't meant to call this method directly? The type parameter inference might make it a little ugly. This is the reason we didn't go with this pattern (builder methods returning constrained generics) for @galvesribeiro's Generic Host PR: because if the user has to specify one generic parameter then they must specify all generic parameters.

I'm not pushing back

Reply:
Yeah that was what I originally did for that PR, but then I saw @ReubenBond's point and makes total sense...

Reply:
Agreed that this can get clumsy in some cases.
I'd not yet come up with a better pattern.
Alternatives?

Reply:
I'd not that I don't think most usages will require users to specify the generics.  See the usage of ConfigureCheckpointer later in this file.  The generics are listed for readability reasons, but are not required.

Though I sued:

    configurator.ConfigureCheckpointer<TConfigurator,AzureTableStreamCheckpointerOptions>(
        EventHubCheckpointerFactory.CreateFactory, configureOptions);

The usage:

    configurator.ConfigureCheckpointer(
        EventHubCheckpointerFactory.CreateFactory, configureOptions);

resolves fine, without need for users to specify configurator type.